tests/CatalogTests/Helpers/UtilsTests.cs

```diff
@@ -109,5 +110,57 @@ private static MemoryStream GetPackageStream()
         {
             return TestHelper.GetStream("Newtonsoft.Json.9.0.2-beta1.nupkg");
         }
+
+        [Fact]
+        public void GetNupkgMetadataWithLicenseUrl_ReturnsLicenseUrl()
+        {
+            // Arrange
+            var stream = TestHelper.GetStream("Newtonsoft.Json.9.0.2-beta1.nupkg");
+            var metadata = Utils.GetNupkgMetadata(stream, packageHash: null);
+            var baseUrl = "http://example/";
+
+            // Actq
+            var graph = Utils.CreateNuspecGraph(metadata.Nuspec, baseUrl, normalizeXml: true);
+            var licenseFileTriples = graph.GetTriplesWithSubjectPredicate(
+                graph.CreateUriNode(new Uri(String.Concat(baseUrl, "newtonsoft.json.9.0.2-beta1.json"))),
+                graph.CreateUriNode(new Uri(String.Concat(Schema.Prefixes.NuGet + "licenseFile")))
+                );
+            var licenseExpressionTriples = graph.GetTriplesWithSubjectPredicate(
+                graph.CreateUriNode(new Uri(String.Concat(baseUrl, "newtonsoft.json.9.0.2-beta1.json"))),
+                graph.CreateUriNode(new Uri(String.Concat(Schema.Prefixes.NuGet + "licenseExpression")))
+                );
+            var licenseUrlTriples = graph.GetTriplesWithSubjectPredicate(
+                graph.CreateUriNode(new Uri(String.Concat(baseUrl, "newtonsoft.json.9.0.2-beta1.json"))),
+                graph.CreateUriNode(Schema.Predicates.LicenseUrl)
+                );
+            
+            // Assert
+            Assert.Equal(0, licenseFileTriples.Count());
+            Assert.Equal(0, licenseExpressionTriples.Count());
+            Assert.Equal(1, licenseUrlTriples.Count());
+        }
+
+        [Theory]
+        [InlineData("TestPackage.LicenseExpression.0.1.0.nupkg", "licenseExpression", "MIT")]
+        [InlineData("TestPackage.LicenseFile.0.1.0.nupkg", "licenseFile", "license.txt")]
+        public void GetNupkgMetadataWithLicenseType_ReturnsLicense(string packageName, string licenseType, string licenseContent)
+        {
+            // Arrange
+            var stream = TestHelper.GetStream(packageName);
+            var metadata = Utils.GetNupkgMetadata(stream, packageHash: null);
+            var baseUrl = "http://example/";
+
+            // Act
+            var graph = Utils.CreateNuspecGraph(metadata.Nuspec, baseUrl, normalizeXml: true);
+            var licenseTriples = graph.GetTriplesWithSubjectPredicate(
+                graph.CreateUriNode(new Uri(String.Concat(baseUrl, "testpackage.license.0.1.0.json"))),
+                graph.CreateUriNode(new Uri(String.Concat(Schema.Prefixes.NuGet, licenseType)))
+                );
+            var result = (LiteralNode)licenseTriples.First().Object;
+            
+            // Assert
+            Assert.Equal(1, licenseTriples.Count());
+            Assert.Equal(licenseContent, result.Value);
```

Comment:
Can we have assertions related to `licenseUrl` too? Basically, I think this should assert that no `licenseUrl` is mixed in here if the package has a `<license>` element

Reply:
Assert licenseUrl now.

Reply:
`licenseUrl` is orthogonal to the license information. In these test cases it is absent because the test packages have no license url specified in nuspec, not because a `license` tag is present. And this is correct behavior. License URL should be "mangled" downstream: catalog2registration and catalog2lucene has to check if there is `licenseExpression` or `licenseFile` present and ignore `licenseUrl` in those cases.

To reflect that in the tests, we probably need to include `licenseUrl` to test packages and check that it is picked up properly and has no effect on presence/value of `licenseFile` and `licenseExpression` properties.

Reply:
Add other test cases now.